addons/web/static/tests/views/list_tests.js

```diff
@@ -1625,6 +1625,40 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('selection box is removed after multi record edition', async function (assert) {
+        assert.expect(6);
+
+        const list = await createView({
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree multi_edit="1"><field name="foo"/><field name="bar"/></tree>',
+        });
+
+        assert.containsN(list, '.o_data_row', 4,
+            "there should be 4 records");
+        assert.containsNone(list.$('.o_cp_buttons'), '.o_list_selection_box',
+            "list selection box should not be displayed");
+
+        // select all records
+        await testUtils.dom.click(list.$('thead .o_list_record_selector input'));
+        assert.containsOnce(list.$('.o_cp_buttons'), '.o_list_selection_box',
+            "list selection box should be displayed");
+        assert.strictEqual(list.renderer.selection.length, list.renderer.state.count,
```

Comment:
you should count the number of data_row with the selector checked, instead of relying on implementation details that could change

Reply:
@aab-odoo 👍